pkg/restore/actions/csi/volumesnapshot_action.go

```diff
@@ -188,13 +143,85 @@ func (p *volumeSnapshotRestoreItemAction) Progress(
 	operationID string,
 	restore *velerov1api.Restore,
 ) (velero.OperationProgress, error) {
-	return velero.OperationProgress{}, nil
+	progress := velero.OperationProgress{}
+
+	// The operationID format should be "vsNamespace/vsName".
+	if operationID == "" || !strings.Contains(operationID, "/") {
+		return progress, riav2.InvalidOperationIDError(operationID)
+	}
+
+	logger := p.log.WithFields(logrus.Fields{
+		"Action":      "VolumeSnapshotRestoreItemAction",
+		"OperationID": operationID,
+		"Namespace":   restore.Namespace,
+	})
+
+	vsc := new(snapshotv1api.VolumeSnapshotContent)
+	vs := new(snapshotv1api.VolumeSnapshot)
+
+	operationIDParts := strings.Split(operationID, "/")
+
+	if err := p.crClient.Get(
+		context.TODO(),
+		crclient.ObjectKey{Namespace: operationIDParts[0], Name: operationIDParts[1]},
+		vs,
+	); err != nil {
+		logger.Errorf("Fail to get the VolumeSnapshot %s", operationID)
+		return progress, err
+	}
+
+	if vs.Spec.Source.VolumeSnapshotContentName == nil {
```

Comment:
This can reuse existing helper function `util.GetVolumeSnapshotContentForVolumeSnapshot`